### PR TITLE
cleanup(pubsub): remove handwritten create subscription from samples.cc

### PR DESCRIPTION
**Pull request by alevenberg**
https://github.com/googleapis/google-cloud-cpp/issues/12987

Exists in subscription_admin_samples.cc now

Also, moved the tag in google3 already.

This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13676)

### CODE REVIEW

File: google/cloud/pubsub/samples/samples.cc [hunk 1]
```diff
@@ -1388,9 +1369,17 @@ void AutoRunAvro(
...
 
-  std::cout << "\nRunning CreateSubscription() sample [avro]" << std::endl;
-  CreateSubscription(subscription_admin_client,
-                     {project_id, avro_topic_id, subscription_id});
+  std::cout << "\nCreate test subscription (" << subscription.subscription_id()
+            << ") [avro]" << std::endl;
+  google::pubsub::v1::Subscription request;
+  request.set_name(subscription.FullName());
+  request.set_topic(topic.FullName());
+  subscription_admin.CreateSubscription(request);
```
Comment (dbolduc) [inline google/cloud/pubsub/samples/samples.cc:1377]: nit: I usually `(void)` these when I am not using the result.

File: google/cloud/pubsub/samples/samples.cc [hunk 2]
```diff
@@ -1460,13 +1448,17 @@ void AutoRunProtobuf(
...
-
-  std::cout << "\nRunning CreateSubscription() sample [avro]" << std::endl;
-  CreateSubscription(subscription_admin_client,
-                     {project_id, proto_topic_id, subscription_id});
+  std::cout << "\nCreate test subscription (" << subscription.subscription_id()
+            << ") [proto]" << std::endl;
+  google::pubsub::v1::Subscription request;
+  request.set_name(subscription.FullName());
+  request.set_topic(topic.FullName());
+  subscription_admin.CreateSubscription(request);
```
Comment (dbolduc) [inline google/cloud/pubsub/samples/samples.cc:1456]: same

File: google/cloud/pubsub/samples/samples.cc [hunk 3]
```diff
@@ -1557,13 +1550,16 @@ void AutoRun(std::vector<std::string> const& argv) {
...
-                     {project_id, topic_id, subscription_id});
-
-  std::cout << "\nRunning CreateSubscription() sample [2]" << std::endl;
-  CreateSubscription(subscription_admin_client,
-                     {project_id, topic_id, subscription_id});
+  std::cout << "\nCreate subscription (" << subscription_id << ")" << std::endl;
+  google::pubsub::v1::Subscription request;
+  request.set_name(subscription.FullName());
+  request.set_topic(topic.FullName());
+  subscription_admin.CreateSubscription(request);
```
Comment (dbolduc) [inline google/cloud/pubsub/samples/samples.cc:1557]: same

File: google/cloud/pubsub/samples/samples.cc [hunk 4]
```diff
@@ -1783,12 +1777,6 @@ void AutoRun(std::vector<std::string> const& argv) {
   DeleteSubscription(subscription_admin_client,
                      {project_id, ordering_subscription_id});
 
-  std::cout << "\nRunning DeleteSubscription() sample [4]" << std::endl;
-  DeleteSubscription(subscription_admin_client, {project_id, subscription_id});
-
-  std::cout << "\nRunning DeleteSubscription() sample [5]" << std::endl;
-  DeleteSubscription(subscription_admin_client, {project_id, subscription_id});
-
   std::cout << "\nRunning DeleteSubscription() sample [6] " << std::endl;
```
Comment (dbolduc) [inline google/cloud/pubsub/samples/samples.cc:1780]: nit: 1.. 2.. 3.. 6.. 7.. 8..

**Review by dbolduc — APPROVED** [review]

File: google/cloud/pubsub/samples/samples.cc (hunk 4 above)
Comment (alevenberg) [reply google/cloud/pubsub/samples/samples.cc:1780]: They'll all be gone eventually 🤷

File: google/cloud/pubsub/samples/samples.cc (hunk 1 above)
Comment (alevenberg) [reply google/cloud/pubsub/samples/samples.cc:1377]:
```suggestion
(void)  subscription_admin.CreateSubscription(request);
```

File: google/cloud/pubsub/samples/samples.cc (hunk 2 above)
Comment (alevenberg) [reply google/cloud/pubsub/samples/samples.cc:1456]:
```suggestion
(void)  subscription_admin.CreateSubscription(request);
```

**Pull request merged**